index.js

```diff
@@ -0,0 +1,222 @@
+import { ready, fire, find, findOne, closest, on, off } from 'domassist';
+import tinybounce from 'tinybounce';
+import aug from 'aug';
+
+const Events = {
+  Enter: 'sticky:enter',
+  Leave: 'sticky:leave'
+};
+
+class Sticky {
+  constructor(el, options = {}) {
+    this.el = findOne(el);
+
+    const {
+      stickyOffset: offset,
+      stickyTarget: target,
+      stickyContainer: container,
+      stickyMatchMedia: matchMedia
+    } = this.el.dataset;
+
+    this.options = aug({
+      offset: parseInt(offset || 0, 10),
+      target,
+      container,
+      matchMedia: window.matchMedia(matchMedia || '(min-width: 768px)')
+    }, options);
+
+    this.targets = [];
+
+    this.options.container = this.options.container ? closest(this.el, this.options.container) : this.el.parentElement;
+
+    find(this.options.target).forEach(targetEl => {
+      const { top, height } = targetEl.getBoundingClientRect();
+
+      this.targets.push({
+        el: targetEl,
+        top: this.scrollTop() + top - height
+      });
+    });
+
+    this.transformTop = 0;
+    this.current = null;
+
+    this.scrollHandler = this.onScroll.bind(this);
+    this.resizeHandler = tinybounce(this.onResize.bind(this), 150);
+
+    this.init();
+  }
+
+  init() {
+    if ('IntersectionObserver' in window) {
+      // eslint-disable-next-line compat/compat
+      this.observer = new IntersectionObserver(this.onIntersect.bind(this));
+      this.observer.observe(this.el);
+    } else {
+      this.calcBounds();
```

Comment:
We should probably call `calcBounds` anyways. It'll likely be called `onResize` anyways.